src/main/java/seedu/mark/logic/commands/ExportCommand.java

```diff
@@ -0,0 +1,53 @@
+package seedu.mark.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import seedu.mark.logic.commands.commandresult.CommandResult;
+import seedu.mark.logic.commands.exceptions.CommandException;
+import seedu.mark.model.Model;
+import seedu.mark.storage.Storage;
+
+/**
+ * Exports Mark's {@code Bookmark} data into a specific file.
+ */
+public class ExportCommand extends Command {
+
+    public static final String COMMAND_WORD = "export";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Exports all of Mark's data to the specified file.\n"
+            + "Parameters: FILENAME\n"
+            + "Example: " + COMMAND_WORD + " savedBookmarks";
+
+    public static final String MESSAGE_EXPORT_SUCCESS = "Bookmarks successfully exported to %1$s";
+    public static final String MESSAGE_EXPORT_FAILURE = "Bookmarks could not be exported";
+
+    private final Path filePath;
+
+    public ExportCommand(Path filePath) {
+        this.filePath = filePath;
+    }
+
+    @Override
+    public CommandResult execute(Model model, Storage storage) throws CommandException {
+        requireNonNull(model);
```

Comment:
I think you can use `requireAllNonNull(mode, storage)` here.

Reply:
Right, I was wondering why my code looked weird. Thanks!